sdk/nodejs/entrypoint/entrypoint.ts

```diff
@@ -37,29 +37,50 @@ async function entrypoint() {
       } else {
         // Invocation
         const fnName = await fnCall.name()
-        const parentJson = await fnCall.parent()
+        const parentJson = JSON.parse(await fnCall.parent())
         const fnArgs = await fnCall.inputArgs()
 
         const args: Args = {}
+        const parentArgs: Args = {}
 
         for (const arg of fnArgs) {
           args[await arg.name()] = await loadArg(await arg.value())
         }
 
+        if (parentJson) {
+          for (const [key, value] of Object.entries(parentJson)) {
+            parentArgs[key] = await loadArg(value as string)
+          }
+        }
+
         await load(files)
 
-        result = await invoke(parentName, fnName, JSON.parse(parentJson), args)
+        result = await invoke(parentName, fnName, parentArgs, args)
 
         // Load ID if it's a Dagger type with an id
-        if (result.id && typeof result.id === "function") {
+        if (result && result.id && typeof result.id === "function") {
           result = await result.id()
         }
+
+        // Load its field ID if there are
+        if (typeof result === "object") {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          for (const [key, value] of Object.entries<any>(result)) {
+            if (value.id && typeof value.id === "function") {
+              result[key] = await value.id()
+            }
+          }
+        }
+      }
+
+      if (result) {
+        result = JSON.stringify(result)
+      } else {
+        result = "null"
       }
```

Comment:
What if `result` is `""`, `0` or other _falsy_ value?
```suggestion
      result = JSON.stringify(result !== undefined ? result : null)
```

Reply:
Fixed in https://github.com/dagger/dagger/pull/6270/commits/691d72fb580b718d93cc5f1dcc88ce46e7d1389f

Reply:
Didn't need `result !== null` since `JSON.stringify(null) == "null"` 🙂